bin/src/command/mod.rs

```diff
@@ -494,24 +494,53 @@ impl CommandServer {
 
             self.workers.push(worker);
         }
+        Ok(())
+    }
+
+    async fn handle_worker_close(&mut self, id: u32) {
+        // put all this in a method handle_worker_close
+        info!("removing worker {}", id);
+
+        if let Some(w) = self.workers.iter_mut().filter(|w| w.id == id).next() {
```

Comment:
is it missing the code we discussed this morning?

Reply:
YES. The experimental code was doing a weird killing/respawning of a single worker in loop. In a matter of seconds, I arrived at worker 405. I could have spent time debugging this, but instead I put it in a separate method, which comes in handy, because: early returns.